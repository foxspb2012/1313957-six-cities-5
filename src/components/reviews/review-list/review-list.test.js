import React from 'react';
import {Provider} from 'react-redux';
import ReviewList from './review-list';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

const NameSpace = {
  USER: `USER`,
  REVIEWS: `REVIEWS`,
};

const reviewsMock = [{
  id: 0,
  date: `2020-10-06`,
  text: `Good`,
  rating: 4,
  user: {
    name: `User`,
    avatar: `img/avatar-angelina.jpg`,
    isPro: false,
    id: 1
  }
},
{
  id: 1,
  date: `2020-10-06`,
  text: `Good`,
  rating: 5,
  user: {
    name: `User`,
    avatar: `img/avatar-max.jpg`,
    isPro: true,
    id: 2
  }
}
];

const mockStateWithAuthorization = {
  [NameSpace.REVIEWS]: {
    reviews: reviewsMock,
    isReviewSending: false,
    error: null
  },
  [NameSpace.USER]: {
    authorizationStatus: `AUTHORIZED`,
    email: null
  }
};

const mockStateWithNoReviews = {
  [NameSpace.REVIEWS]: {
    reviews: [],
    isReviewSending: false,
    error: null
  },
  [NameSpace.USER]: {
    authorizationStatus: `AUTHORIZED`,
    email: null
  }
};

describe(`Should ReviewList render correctly`, () => {
  it(`Should ReviewList render correctly with reviews`, () => {
    const mockStore = configureStore();
    const store = mockStore(mockStateWithAuthorization);
    const tree = renderer
      .create(
          <Provider store={store}>
            <ReviewList/>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should ReviewList render correctly when no reviews`, () => {
    const mockStore = configureStore();
    const store = mockStore(mockStateWithNoReviews);
    const tree = renderer
      .create(
          <Provider store={store}>
            <ReviewList/>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
