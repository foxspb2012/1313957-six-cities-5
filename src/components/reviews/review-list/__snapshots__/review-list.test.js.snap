// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ReviewList render correctly Should ReviewList render correctly when no reviews 1`] = `
<ul
  className="reviews__list"
/>
`;

exports[`Should ReviewList render correctly Should ReviewList render correctly with reviews 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="User photo"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-angelina.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        User
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="rating__stars reviews__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Good
      </p>
      <time
        className="reviews__time"
        dateTime="2020-10-06"
      >
        October 2020
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="User photo"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        User
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="rating__stars reviews__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Good
      </p>
      <time
        className="reviews__time"
        dateTime="2020-10-06"
      >
        October 2020
      </time>
    </div>
  </li>
</ul>
`;
